infrastructure/index.ts

```diff
@@ -804,6 +810,47 @@ const cdn = new aws.cloudfront.Distribution(
     },
 );
 
+// Configure bucket policy to allow CloudFront OAC access and deny direct access
+// This must be created after the distribution so we can reference its ARN
+const originBucketPolicy = new aws.s3.BucketPolicy("origin-bucket-policy", {
+    bucket: originBucket.bucket,
+    policy: pulumi.all([originBucket.arn, cdn.arn, aws.getCallerIdentity()])
+        .apply(([bucketArn, distributionArn, awsCallerIdentityResult]) => JSON.stringify({
+            Version: "2012-10-17",
+            Statement: [
+                {
+                    Sid: "AllowCloudFrontServicePrincipalReadOnly",
+                    Effect: "Allow",
+                    Principal: {
+                        Service: "cloudfront.amazonaws.com",
+                    },
+                    Action: ["s3:GetObject", "s3:ListBucket"],
+                    Resource: [bucketArn, `${bucketArn}/*`],
+                    Condition: {
+                        StringEquals: {
+                            "AWS:SourceArn": distributionArn,
+                        },
+                    },
+                },
+                {
+                    Sid: "DenyDirectListBucket",
```

Comment:
Deny's can be rather dangerous and I think should be expressly configured so there's not a risk they drift on context.

The allow as you have it should be sufficient in preventing list access to non-admins.  I would leave it off unless there are specific requirements for a deny.

Reply:
Approved to not be a blocker since it was also already there and it's really just a housecleaning item.